source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/admin.js

```diff
@@ -70,12 +78,63 @@
             CurrentPosition: this.currentPosition,
             Seniority: this.seniority,
             FullName: this.fullName,
-            Period: this.period
+            Period: this.period,
+            IsValid: this.isValid()
         };
     }
-
     var viewmodel = new PeriodCreation();
 
+    $('#generate-evaluation-button').on('click', function () {
+        if (viewmodel.isValid()) {
+            postEvaluationsForGeneration({
+                onSuccess: function (result) {
+                    showEvaluationGenerationResult(true, result.amountGenerated);
+                },
+                onError: function () {
+                    alert('Fallo interno. Por favor recargue la página.');
+                }
+            });
+        } else {
+            showEvaluationGenerationResult(false);
+        }
+
+        function postEvaluationsForGeneration(callbacks) {
+            var model = viewmodel.toJs();
+            var modelFiltered = { Employees: _.filter(model.Employees, function (e) { return e.ResponsibleId && !e.Period; }) };
+            $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
+                type: "POST",
+                dataType: 'json',
+                contentType: 'application/json; charset=utf-8',
+                data: JSON.stringify(modelFiltered),
+                complete: function (response) {
+                    var result = { amountGenerated: parseInt(response.responseText) }
+                    if (typeof callbacks.onSuccess === 'function') { callbacks.onSuccess(result); }
+                }
+            });
+            error = callbacks.onError;
+        };
+
+        function showEvaluationGenerationResult(wereGenerated, amountGenerated) {
```

Comment:
That's exactly right! You can declare it outside and refer to it by name.

Note that the problem is not with the contents of the click method, but that you're doing this:

```javascript
$('#generate-evaluation-button').on('click', function () {
  // ...
  showEvaluationGenerationResult();

  // ...
  function showEvaluationGenerationResult() { // this funcion is inside the other anonymous function for no reason, it's re-declared each time that the anonymous function is invoked
    // ...
  }
});
```

When you should do this:

```javascript
function showEvaluationGenerationResult() { // this function is now one and gets only used instead of regenerated
  // ...
}

$('#generate-evaluation-button').on('click', function () {
  // ...
  showEvaluationGenerationResult();
});
```